ui-ngx/src/app/core/services/utils.service.ts

```diff
@@ -479,4 +488,30 @@ export class UtilsService {
       return defaultValue;
     }
   }
+
+  public getEntityIdFromDatasource(dataSource: Datasource) {
+    return { id: dataSource.entityId, entityType: dataSource.entityType};
+  }
+
+  public createTelemetrySubscriber(ctx: WidgetContext,
+                                   entityId?: EntityId,
+                                   type: TelemetryType = LatestTelemetry.LATEST_TELEMETRY,
+                                   keys: string[] = null): TelemetrySubscriber {
+    if (!entityId && ctx.datasources.length > 0) {
+        entityId = this.getEntityIdFromDatasource(ctx.datasources[0]);
+    }
+    return TelemetrySubscriber.createEntityAttributesSubscription(ctx.telemetryWsService, entityId, type, ctx.ngZone, keys);
+  }
+
+  public subscribeToEntityTelemetry(ctx: WidgetContext,
+                                    entityId?: EntityId,
+                                    type: TelemetryType = LatestTelemetry.LATEST_TELEMETRY,
+                                    keys: string[] = null): Observable<Array<AttributeData>> {
+    ctx.telemetrySubscriber = this.createTelemetrySubscriber(ctx, entityId, type, keys);
+    ctx.telemetrySubscriber.subscribe();
```

Comment:
This code can cause a memory leak:
1. If the subscription is stored in the context, the user may not know exactly where to unsubscribe. As a solution, you need to automatically unsubscribe when the widget component is destroyed.
2. The user can call this function several times, and the previous subscription will not be cleared. As a solution, you can store all subscriptions in an array and clear them when the widget is destroyed or return an observable to the user